src/map.js

```diff
@@ -4,13 +4,17 @@ import mapInteractions from './mapinteractions';
 
 const Map = (options = {}) => {
   const interactions = mapInteractions({ target: options.target, mapInteractions: options.pageSettings && options.pageSettings.mapInteractions ? options.pageSettings.mapInteractions : {} });
+  const mapConfig = Object.assign({}, options);
+  delete mapConfig.layers;
+  delete mapConfig.styles;
+  delete mapConfig.source;
```

Comment:
Can this be done without first assign and then delete? Layers and styles can be quite large so wasting memory perhaps.

Reply:
I suppose a bit more verbose solution is to loop through an array of the keys and skip some of them.

Reply:
Rewrote this a bit